native/src/CMakeLists.txt

```diff
@@ -38,14 +41,40 @@ else(DEFINED ENV{RAFT_PATH})
   set(RAFT_INCLUDE_DIR ${RAFT_GIT_DIR}/src/raft/cpp/include CACHE STRING "RAFT include variable")
 endif(DEFINED ENV{RAFT_PATH})
 
+
+
+#################################################################################################
+# - CPM -----------------------------------------------------------------------------------------
+
+set(CPM_DOWNLOAD_VERSION 0.27.2)
+set(CPM_DOWNLOAD_LOCATION "${CMAKE_BINARY_DIR}/cmake/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
+
+if(NOT (EXISTS ${CPM_DOWNLOAD_LOCATION}))
+    message(STATUS "Downloading CPM.cmake")
+    file(DOWNLOAD https://github.com/TheLartians/CPM.cmake/releases/download/v${CPM_DOWNLOAD_VERSION}/CPM.cmake ${CPM_DOWNLOAD_LOCATION})
+endif()
+
+include(${CPM_DOWNLOAD_LOCATION})
+#################################################################################################
+
+# pull cuDF sources, to use jni_utils.hpp
+# cmake options should be added here for CI build.
+CPMAddPackage(NAME cudf
+        VERSION         "21.12.00"
+        GIT_REPOSITORY  https://github.com/rapidsai/cudf.git
+        GIT_TAG         branch-21.12
```

Comment:
no cmake options because I have installed cuDF via Conda, it's used for local dev only. Will add options here for official CI.